superset/databases/commands/tables.py

```diff
@@ -0,0 +1,113 @@
+    DatabaseNotFoundError,
+    DatabaseTablesUnexpectedError,
+)
+from superset.databases.dao import DatabaseDAO
+from superset.exceptions import SupersetException
+from superset.extensions import db, security_manager
+from superset.models.core import Database
+from superset.utils.core import DatasourceName
+
+logger = logging.getLogger(__name__)
+
+
+class TablesDatabaseCommand(BaseCommand):
+    _model: Database
+
+    def __init__(self, db_id: int, schema_name: str, force: bool):
+        self._db_id = db_id
+        self._schema_name = schema_name
+        self._force = force
+
+    def run(self) -> Dict[str, Any]:
+        self.validate()
+        try:
+            tables = security_manager.get_datasources_accessible_by_user(
+                database=self._model,
+                schema=self._schema_name,
+                datasource_names=sorted(
+                    DatasourceName(*datasource_name)
+                    for datasource_name in self._model.get_all_table_names_in_schema(
+                        schema=self._schema_name,
+                        force=self._force,
+                        cache=self._model.table_cache_enabled,
+                        cache_timeout=self._model.table_cache_timeout,
+                    )
+                ),
+            )
+
+            views = security_manager.get_datasources_accessible_by_user(
+                database=self._model,
+                schema=self._schema_name,
+                datasource_names=sorted(
+                    DatasourceName(*datasource_name)
+                    for datasource_name in self._model.get_all_view_names_in_schema(
+                        schema=self._schema_name,
+                        force=self._force,
+                        cache=self._model.table_cache_enabled,
+                        cache_timeout=self._model.table_cache_timeout,
+                    )
+                ),
+            )
+
+            extra_dict_by_name = {
+                table.name: table.extra_dict
+                for table in (
+                    db.session.query(SqlaTable).filter(
+                        SqlaTable.database_id == self._model.id,
+                        SqlaTable.schema == self._schema_name,
+                    )
+                ).all()
+            }
+
+            options = sorted(
+                [
+                    {
+                        "value": table.table,
+                        "type": "table",
+                        "extra": extra_dict_by_name.get(table.table, None),
+                    }
+                    for table in tables
+                ]
+                + [
+                    {
+                        "value": view.table,
+                        "type": "view",
+                    }
+                    for view in views
+                ],
+                key=lambda item: item["value"],
+            )
+
+            payload = {"tableLength": len(tables) + len(views), "options": options}
```

Comment:
If `tableLength` is a count of all records on the payload, we are using `count` on all other API endpoints